### ISSUE TITLE
Feature: Animations

### ISSUE THREAD

**Issue by bdlukaa** [leoafarias/mix#4]
Provide apis for animations. Something like:

```dart
AnimatedMix(Mix(
  bgColor(Colors.green),
))
```

Whenever the Mix is changed, it should be animated.

**Comment by bdlukaa** [comment]
Or maybe something like
```
Mix(
  bgColor(Colors.green).animated(Duration(milliseconds: 300)), // ofc there is a default duration
),
```

**Comment by bdlukaa** [comment]
I think I know how to do this. Every attribute will have the `animationDuration` and `animationCurve`, always with null as default.

Whenever the `animated` function is called, it sets the animation duration and curve, and the widget will need to handle that. The API will be same as presented above.

**Comment by leoafarias** [comment]
@bdlukaa quick question, when would the animation take place. Is that when it's doing a transition between Mix?

The API on AnimatedContainer allows you to pick properties that are animated. I wonder if a higher level class `AnimatedMix`, would allow to define the animations, and maybe further configuration like duration, curve and so on.

**Comment by bdlukaa** [comment]
I don't think `AnimatedMix` would be a good thing. It'd cause much confusion and the simplicity of mix would go away.

On the other hand, I also don't think that `bgColor(Colors.green).animated(Duration(milliseconds: 300))` would be good too. It'd cause so much boilerplate code when multiple properties needs to be animated (like an `AnimatedContainer`). What we could have is:

1. The following approach will animate separated attributes

```dart
Mix(
  bgColor(Colors.green).animated(Duration(milliseconds: 300)),
),
```

2. The following approach will animate all the attributes

```dart
Mix(
  animated(Duration(milliseconds: 300)),
  bgColor(Colors.green),
),
```

3. The following approach will animate only the selected attributes

```dart
Mix(
  animated(Duration(milliseconds: 300)),
  bgColor(Colors.green).animated(),
  textColor(Colors.white),
),
```

**Comment by leoafarias** [comment]
I will add also the ability to apply Mixes and account for that. We probably would have to decorate these attributes with an animated attribute that has those `animated` properties to keep composability.

```dart
final simpleMix = Mix(
  animated(Duration(milliseconds: 300)),
  bgColor(Colors.green).animated(),
  textColor(Colors.white),
),

Mix(
  apply(simpleMix),
),

```

**Comment by bdlukaa** [comment]
Fixed on #7, #9 and #10
